tests/test_currency_network_interests.py

```diff
@@ -620,3 +620,46 @@ def test_interests_over_creditline_is_usable(
     assert contract.functions.balance(accounts[0], accounts[1]).call() == pytest.approx(
         10201, abs=2
     )
+
+
+def test_correct_balance_update_event_on_interest_rate_change(
+    chain, currency_network_contract_default_interests, accounts
+):
+    contract = currency_network_contract_default_interests
+    current_time = int(time.time())
+    chain.time_travel(current_time + 10)
+
+    # Set trustline
+    contract.functions.updateTrustline(accounts[0], 10000, 10000, 100, 100).transact(
+        {"from": accounts[1]}
+    )
+    contract.functions.updateTrustline(accounts[1], 10000, 10000, 100, 100).transact(
+        {"from": accounts[0]}
+    )
+    contract.functions.transfer(accounts[0], 10000, 0, [accounts[0]]).transact(
+        {"from": accounts[1]}
+    )
+
+    # Time travel
+    chain.time_travel(current_time + SECONDS_PER_YEAR)
+
+    # Update trustline
+    contract.functions.updateTrustline(accounts[0], 11000, 11000, 100, 100).transact(
+        {"from": accounts[1]}
+    )
+    contract.functions.updateTrustline(accounts[1], 11000, 11000, 100, 100).transact(
+        {"from": accounts[0]}
+    )
+
+    # Check event
+    events = contract.events.BalanceUpdate.createFilter(fromBlock=0).get_all_entries()
```

Comment:
What you are checking there is the BalanceUpdate event due to the transfer of line 639. 

I guess what you meant to do is to check the BalanceUpdate event after updating the interest rate and not the credit limits as you do, so the event due to line 650.

Concretely, you do not test what event you get from `events[-1]` and as far as I can tell you receive the wrong event.

Reply:
No, I just check the last Balance Update and if that has the same value as the current balance in the state. I do not care from where it comes, only that the value in it is the same as in the state. 

If it would be the one from the last transfer it should be wrong as in between there is a year of interests.

Reply:
Ah I see, I dont change the interest rate, fixing this.